pytorch_lightning/plugins/collective/horovod_collective.py

```diff
@@ -0,0 +1,107 @@
+# Copyright The PyTorch Lightning team.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+from typing import Any, List, Optional, Union
+
+import torch
+
+from pytorch_lightning.plugins.collective import Collective
+from pytorch_lightning.utilities import _HOROVOD_AVAILABLE
+from pytorch_lightning.utilities.distributed import distributed_available
+from pytorch_lightning.utilities.distributed import group as dist_group
+from pytorch_lightning.utilities.distributed import ReduceOp
+
+if _HOROVOD_AVAILABLE:
+    import horovod.torch as hvd
+
+
+class HorovodCollective(Collective):
+    """Collective interface for Horovod training type plugins."""
+
+    def __init__(
+        self,
+        on_gpu: Optional[bool] = False,
+        local_rank: int = 0,
+    ) -> None:
+        self.on_gpu = on_gpu
```

Comment:
Let's raise here if `_HOROVOD_AVAILABLE ` isn't available.

Reply:
Should we raise it in HorovodPlugin then?
I think current plugins has consistent behavior that : doesn't have raise exception when module/device doesn't exist (eg: hvd, tpu, deepspeed), should we do a improvement have raise exception for all of them?

Current deepspeed
```
if _DEEPSPEED_AVAILABLE:
    import deepspeed
```


Current tpu
```
if _TPU_AVAILABLE:
    import torch_xla.core.xla_env_vars as xenv
    import torch_xla.core.xla_model as xm
    import torch_xla.distributed.xla_multiprocessing as xmp
    from torch_xla.core.xla_model import rendezvous
    from torch_xla.distributed.parallel_loader import MpDeviceLoader
else:
    xm, xmp, MpDeviceLoader, rendezvous = [None] * 4
```
If None.function will through exception anyway. Should we add check in each plugin?

Reply:
IMO we could raise it here too. its a good check in case someone uses these collectives independently of the training type (for whatever reason)

Reply:
Will create a issue add raise exception for xla and horovod in training type plugin too